pulp_smash/tests/rpm/api_v2/test_upload_publish.py

```diff
@@ -33,47 +33,36 @@
 from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
 
 
-class UploadDrpmTestCase(utils.BaseAPITestCase):
+class UploadDrpmTestCase(unittest.TestCase):
     """Test whether one can upload a DRPM into a repository.
 
+    Specifically, this method does the following:
+
+    1. Create a yum repository.
+    2. Upload a DRPM into the repository.
+    3. Search for all content units in the repository.
+
     This test case targets `Pulp Smash #336
     <https://github.com/PulpQE/pulp-smash/issues/336>`_
     """
 
-    @classmethod
-    def setUpClass(cls):
-        """Import a DRPM into a repository and search it for content units.
-
-        Specifically, this method does the following:
-
-        1. Create a yum repository.
-        2. Upload a DRPM into the repository.
-        3. Search for all content units in the repository.
-        """
-        super(UploadDrpmTestCase, cls).setUpClass()
-        if selectors.bug_is_untestable(1806, cls.cfg.version):
-            raise unittest.SkipTest('https://pulp.plan.io/issues/1806')
-        client = api.Client(cls.cfg)
+    def test_all(self):
+        """Import a DRPM into a repository and search it for content units."""
+        cfg = config.get_config()
+        if selectors.bug_is_untestable(1806, cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/1806')
+        client = api.Client(cfg)
         repo = client.post(REPOSITORY_PATH, gen_repo()).json()
-        cls.resources.add(repo['_href'])
+        self.addCleanup(client.delete, repo['_href'])
         drpm = utils.http_get(DRPM_UNSIGNED_URL)
-        utils.upload_import_unit(cls.cfg, drpm, {'unit_type_id': 'drpm'}, repo)
-        cls.units = utils.search_units(cls.cfg, repo, {}, api.safe_handler)
-
-    def test_status_code_units(self):
-        """Verify the HTTP status code for repo units response."""
-        self.assertEqual(self.units.status_code, 200)
+        utils.upload_import_unit(cfg, drpm, {'unit_type_id': 'drpm'}, repo)
+        units = utils.search_units(cfg, repo, {}, api.safe_handler) .json()
```

Comment:
Here's the definition of `search_units`:

```python
def search_units(cfg, repo, criteria=None, response_handler=None):
    # snip!
    if criteria is None:
        criteria = {}
    if response_handler is None:
        response_handler = api.json_handler
    return api.Client(cfg, response_handler).post(
        urljoin(repo['_href'], 'search/units/'),
        {'criteria': criteria},
    )
```

Notice the default values given to `criteria` and `response_handler`? With that in mind, this line of code can be rewritten as:

```python
units = utils.search_units(cfg, repo)
```